blog_article/blog_article.adoc

```diff
@@ -0,0 +1,161 @@
+For AMQP 1.0 messages that should be read by AMQP 0.9.1 clients, it should be considered that AMQP 1.0 uses
+the so-called AMQP-encoding. This means that all messages that are sent with AMQP 1.0 are encoded beforehand.
+The grammar of this encoding is specified within the AMQP 1.0 protocol. However, this grammar can be implemented
+differently by each client. +
+The consequences will become clear when looking at an example:
+
+* Sent message using SwiftMQ client (AMQP 1.0): M1
+* Received message using RabbitMQ client (AMQP 0.9.1): �w�M1
+
+* Sent message using Qpid JMS client (AMQP 1.0): M1
+* Received message using RabbitMQ client (AMQP 0.9.1): Sw�M1
+
+The examples show that there are some additional bytes appended in front of the messages. The reason for this is the
+AMQP-encoding that is used by the message producers. Since this encoding seems not to be implemented the same for
+every client, the AMQP-plugin cannot cut those additional bytes off. This issue was also discussed in the RabbitMQ
+https://groups.google.com/g/rabbitmq-users/c/QBsc_wCuTBs/m/xh84gj1hAwAJ[mailing list].
+Despite the slightly different implementation of the encoding and decoding of the clients, it is possible to send
+and receive AMQP 1.0 messages with different AMQP 1.0 client libraries without constraints.
+
+The AMQP 1.0 protocol also includes a possibility to send unencoded messages. Data can be sent as payload of the
+_data field_ of an AMQP 1.0 message. Data included in this _data field_ will not be encoded and therefore sent as plain
+bytes. This ensures that clients who do not speak AMQP-encoding can speak with AMQP 1.0 communication partners. +
+To use this functionality, it needs to be explicitly implemented within the code. Therefore, it needs to be known at
+the time of the implementation if there are or will be communication partners who do not speak AMQP 1.0.
+
+SwiftMQ clients need to explicitly check if data was sent within the data field.
+If messages are sent in AMQP 0.9.1 format, the RabbitMQ plugin will forward these messages to AMQP 1.0 clients as
+part of the message data field. In contrast to SwiftMQ, the Qpid JMS client library checks for payload in the data
+field internally. Therefore, this check has only to be done when using SwiftMQ.
+
+
+== Summary
+When using RabbitMQ as a broker, messages can be exchanged in AMQP 0.9.1 and AMQP 1.0 format.
+Nevertheless, this can only be done when some constraints are considered at the time of the implementation.
+Therefore, when adding new components to existing systems, the question remains if these considerations can always
+be taken into account. And if this can be taken into account, it may still be easier to stay with one AMQP specification.
+At this point, it can also be added that at least for RabbitMQ, AMQP 0.9.1 remains supported in the long term. +
+AMQP 1.0 was introduced to become a new standard in interoperability between messaging middleware. Still, many systems
```

Comment:
I would mention (maybe already earlier in the previous section), that it could be an advantage to not use AMQP 1.0 as message format, but rather Avro or Protobuf. They are much more flexible than AMQP 1.0 format for data and provide much more flexibility. Therefore I think that it is always a good approach to not use AMQP 1.0 (because its dedicated to AMQP 1.0) and instead directly start with an widely used format like plain json, avro, protobuf, ...